examples/all-clusters-app/esp32/main/DeviceWithDisplay.cpp

```diff
@@ -182,20 +209,69 @@ class EditAttributeListModel : public TouchesMatterStackModel
                 ESP_LOGI(TAG, "Humidity changed to : %d", n);
                 app::Clusters::RelativeHumidityMeasurement::Attributes::MeasuredValue::Set(1, static_cast<int16_t>(n * 100));
             }
-            else if (name == "OccupiedCoolingSetpoint")
+            else if (name == "CoolSetpoint")
             {
-                ESP_LOGI(TAG, "OccupiedCoolingSetpoint changed to : %d", n);
+                // update the occupied cooling setpoint for hardcoded endpoint 1
+                ESP_LOGI(TAG, "Occupied Cooling Setpoint changed to : %d", n);
                 app::Clusters::Thermostat::Attributes::OccupiedCoolingSetpoint::Set(1, static_cast<int16_t>(n * 100));
             }
-            else if (name == "OccupiedHeatingSetpoint")
+            else if (name == "HeatSetpoint")
             {
-                ESP_LOGI(TAG, "OccupiedHeatingSetpoint changed to : %d", n);
+                // update the occupied heating setpoint for hardcoded endpoint 1
+                ESP_LOGI(TAG, "Occupied Heating Setpoint changed to : %d", n);
                 app::Clusters::Thermostat::Attributes::OccupiedHeatingSetpoint::Set(1, static_cast<int16_t>(n * 100));
             }
             else if (name == "SystemMode")
             {
-                ESP_LOGI(TAG, "SystemMode changed to : %d", n);
-                app::Clusters::Thermostat::Attributes::OccupiedHeatingSetpoint::Set(1, n);
+                uint8_t mode = n % 5;
+                // Update the system mode here for hardcoded endpoint 1
+               if (isValidThermostatSystemMode(mode))
+               {
+                    ESP_LOGI(TAG, "System Mode changed to : %d", mode);
+                    app::Clusters::Thermostat::Attributes::SystemMode::Set(1, static_cast<uint8_t>(mode));
+                    // If system mode is auto set running mode to off otherwise set it to what the system mode is set to
+                    if (mode == ThermostatSystemMode::kAuto)
+                    {
+                        app::Clusters::Thermostat::Attributes::ThermostatRunningMode::Set(1, static_cast<uint8_t>(ThermostatRunningMode::kOff));
```

Comment:
Why is this casting to int16?

Reply:
ah its incorrect. should be uint8_t

Reply:
not sure. I understand. i am using uint8_t.